Index: intltool-extract.in
===================================================================
--- intltool-extract.in	(revision 758)
+++ intltool-extract.in	(working copy)
@@ -31,8 +31,8 @@
 
 ## Release information
 my $PROGRAM      = "intltool-extract";
-my $PACKAGE      = "@PACKAGE@";
-my $VERSION      = "@VERSION@";
+my $PACKAGE      = "intltool";
+my $VERSION      = "0.40.3";
 
 ## Loaded modules
 use strict; 
Index: intltool.m4
===================================================================
--- intltool.m4	(revision 758)
+++ intltool.m4	(working copy)
@@ -1,3 +1,6 @@
+# This is intltool.m4 from intltool-0.40, fixed back so that user doesn't
+# have to have it installed.
+
 ## intltool.m4 - Configure intltool for the target system. -*-Shell-script-*-
 ## Copyright (C) 2001 Eazel, Inc.
 ## Author: Maciej Stachowiak <mjs@noisehavoc.org>
@@ -36,24 +39,7 @@ case "$am__api_version" in
     ;;
 esac
 
-if test -n "$1"; then
-    AC_MSG_CHECKING([for intltool >= $1])
-
-    INTLTOOL_REQUIRED_VERSION_AS_INT=`echo $1 | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
-    INTLTOOL_APPLIED_VERSION=`intltool-update --version | head -1 | cut -d" " -f3`
-    [INTLTOOL_APPLIED_VERSION_AS_INT=`echo $INTLTOOL_APPLIED_VERSION | awk -F. '{ print $ 1 * 1000 + $ 2 * 100 + $ 3; }'`
-    ]
-    AC_MSG_RESULT([$INTLTOOL_APPLIED_VERSION found])
-    test "$INTLTOOL_APPLIED_VERSION_AS_INT" -ge "$INTLTOOL_REQUIRED_VERSION_AS_INT" ||
-	AC_MSG_ERROR([Your intltool is too old.  You need intltool $1 or later.])
-fi
-
-AC_PATH_PROG(INTLTOOL_UPDATE, [intltool-update])
-AC_PATH_PROG(INTLTOOL_MERGE, [intltool-merge])
-AC_PATH_PROG(INTLTOOL_EXTRACT, [intltool-extract])
-if test -z "$INTLTOOL_UPDATE" -o -z "$INTLTOOL_MERGE" -o -z "$INTLTOOL_EXTRACT"; then
-    AC_MSG_ERROR([The intltool scripts were not found. Please install intltool.])
-fi
+## Do not check version at all, since it's all kept in the repo
 
   INTLTOOL_DESKTOP_RULE='%.desktop:   %.desktop.in   $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; LC_ALL=C $(INTLTOOL_MERGE) -d -u -c $(top_builddir)/po/.intltool-merge-cache $(top_srcdir)/po $< [$]@' 
 INTLTOOL_DIRECTORY_RULE='%.directory: %.directory.in $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; LC_ALL=C $(INTLTOOL_MERGE) -d -u -c $(top_builddir)/po/.intltool-merge-cache $(top_srcdir)/po $< [$]@' 
@@ -110,6 +96,11 @@ if test -z "$xgversion" -o -z "$mmversio
     AC_MSG_ERROR([GNU gettext tools not found; required for intltool])
 fi
 
+# Use the tools built into the package, not the ones that are installed.
+AC_SUBST(INTLTOOL_EXTRACT, '$(top_builddir)/intltool-extract')
+AC_SUBST(INTLTOOL_MERGE, '$(top_builddir)/intltool-merge')
+AC_SUBST(INTLTOOL_UPDATE, '$(top_builddir)/intltool-update')
+
 AC_PATH_PROG(INTLTOOL_PERL, [perl])
 if test -z "$INTLTOOL_PERL"; then
    AC_MSG_ERROR([perl not found; required for intltool])
@@ -126,6 +117,42 @@ if test "x$2" != "xno-xml"; then
    fi
 fi
 
+dnl The following is very similar to
+dnl
+dnl	AC_CONFIG_FILES([intltool-extract intltool-merge intltool-update])
+dnl
+dnl with the following slight differences:
+dnl  - the *.in files are in ac_aux_dir,
+dnl  - if the file haven't changed upon reconfigure, it's not touched,
+dnl  - the evaluation of the third parameter enables a hack which computes
+dnl    the actual value of $libdir,
+dnl  - the user sees "executing intltool commands", instead of
+dnl    "creating intltool-extract" and such.
+dnl
+dnl Nothing crucial here, and we could use AC_CONFIG_FILES, if there were
+dnl a reason for it.
+
+AC_CONFIG_COMMANDS([intltool], [
+
+for file in intltool-extract intltool-merge intltool-update; do
+  sed -e "s|@INTLTOOL_EXTRACT@|`pwd`/intltool-extract|g" \
+      -e "s|@INTLTOOL_LIBDIR@|${INTLTOOL_LIBDIR}|g" \
+      -e "s|@INTLTOOL_PERL@|${INTLTOOL_PERL}|g" \
+	< ${ac_aux_dir}/${file}.in > ${file}.out
+  if cmp -s ${file} ${file}.out 2>/dev/null; then
+    rm -f ${file}.out
+  else
+    mv -f ${file}.out ${file}
+  fi
+  chmod ugo+x ${file}
+  chmod u+w ${file}
+done
+
+],
+[INTLTOOL_PERL='${INTLTOOL_PERL}' ac_aux_dir='${ac_aux_dir}'
+prefix="$prefix" exec_prefix="$exec_prefix" INTLTOOL_LIBDIR="$libdir" 
+INTLTOOL_EXTRACT='${INTLTOOL_EXTRACT}'])
+
 # Substitute ALL_LINGUAS so we can use it in po/Makefile
 AC_SUBST(ALL_LINGUAS)
 
Index: intltool-update.in
===================================================================
--- intltool-update.in	(revision 758)
+++ intltool-update.in	(working copy)
@@ -1,4 +1,4 @@
-#!@INTLTOOL_PERL@ -w
+#!@INTLTOOL_PERL@
 # -*- Mode: perl; indent-tabs-mode: nil; c-basic-offset: 4  -*-
 
 #
@@ -30,8 +30,8 @@
 
 ## Release information
 my $PROGRAM = "intltool-update";
-my $VERSION = "@VERSION@";
-my $PACKAGE = "@PACKAGE@";
+my $VERSION = "0.40.3";
+my $PACKAGE = "intltool";
 
 ## Loaded modules
 use strict;
@@ -442,7 +442,7 @@ sub FindLeftoutFiles
 		{
 		    if (!s-$match([^\\]|\\.)*?$match-QUOTEDTEXT-)
 		    {
-			warn "mismatched quotes at line $. in $file\n";
+			# warn "mismatched quotes at line $. in $file\n";
 			s-$match.*--;
 		    }
 		}
Index: intltool-merge.in
===================================================================
--- intltool-merge.in	(revision 758)
+++ intltool-merge.in	(working copy)
@@ -34,8 +34,8 @@
 
 ## Release information
 my $PROGRAM = "intltool-merge";
-my $PACKAGE = "@PACKAGE@";
-my $VERSION = "@VERSION@";
+my $PACKAGE = "intltool";
+my $VERSION = "0.40.3";
 
 ## Loaded modules
 use strict; 
