diff -r 1a453a5a7552 src/cairo-win32-surface.c
--- a/src/cairo-win32-surface.c	Sun Jul 29 00:55:21 2012 -0700
+++ b/src/cairo-win32-surface.c	Sun Jul 29 01:03:22 2012 -0700
@@ -92,11 +92,11 @@
 			 NULL,
 			 last_error,
 			 MAKELANGID (LANG_NEUTRAL, SUBLANG_DEFAULT),
-			 (LPSTR) &lpMsgBuf,
+			 (LPWSTR) &lpMsgBuf,
 			 0, NULL)) {
 	fprintf (stderr, "%s: Unknown GDI error", context);
     } else {
-	fwprintf (stderr, "%S: %s", context, (char *)lpMsgBuf);
+	fwprintf (stderr, L"%S: %s", context, lpMsgBuf);
 
 	LocalFree (lpMsgBuf);
     }
