diff -r 42fc00f7cdcb gettext-tools/gnulib-lib/Makefile.gnulib
--- a/gettext-tools/gnulib-lib/Makefile.gnulib	Mon Aug 06 02:08:14 2012 -0700
+++ b/gettext-tools/gnulib-lib/Makefile.gnulib	Mon Aug 06 02:09:11 2012 -0700
@@ -816,42 +816,42 @@
 	mv -f $@-t $@
 
 glib/ghash.h: glib/ghash.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/ghash.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/glist.h: glib/glist.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/glist.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gprimes.h: glib/gprimes.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gprimes.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gstrfuncs.h: glib/gstrfuncs.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gstrfuncs.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gstring.h: glib/gstring.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gstring.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gtypes.h: glib/gtypes.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gtypes.in.h; \
 	} > $@-t
@@ -935,329 +935,329 @@
 BUILT_SOURCES += $(LIBXML_H)
 
 libxml/DOCBparser.h: libxml/DOCBparser.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/DOCBparser.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/HTMLparser.h: libxml/HTMLparser.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/HTMLparser.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/HTMLtree.h: libxml/HTMLtree.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/HTMLtree.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/SAX.h: libxml/SAX.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/SAX.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/SAX2.h: libxml/SAX2.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/SAX2.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/c14n.h: libxml/c14n.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/c14n.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/catalog.h: libxml/catalog.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/catalog.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/chvalid.h: libxml/chvalid.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/chvalid.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/debugXML.h: libxml/debugXML.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/debugXML.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/dict.h: libxml/dict.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/dict.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/encoding.h: libxml/encoding.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/encoding.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/entities.h: libxml/entities.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/entities.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/globals.h: libxml/globals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/globals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/hash.h: libxml/hash.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/hash.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/list.h: libxml/list.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/list.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/nanoftp.h: libxml/nanoftp.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/nanoftp.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/nanohttp.h: libxml/nanohttp.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/nanohttp.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/parser.h: libxml/parser.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/parser.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/parserInternals.h: libxml/parserInternals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/parserInternals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/pattern.h: libxml/pattern.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/pattern.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/relaxng.h: libxml/relaxng.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/relaxng.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/schemasInternals.h: libxml/schemasInternals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/schemasInternals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/schematron.h: libxml/schematron.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/schematron.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/threads.h: libxml/threads.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/threads.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/tree.h: libxml/tree.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/tree.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/uri.h: libxml/uri.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/uri.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/valid.h: libxml/valid.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/valid.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xinclude.h: libxml/xinclude.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xinclude.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xlink.h: libxml/xlink.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xlink.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlIO.h: libxml/xmlIO.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlIO.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlautomata.h: libxml/xmlautomata.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlautomata.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlerror.h: libxml/xmlerror.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlerror.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlexports.h: libxml/xmlexports.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlexports.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlmemory.h: libxml/xmlmemory.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlmemory.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlmodule.h: libxml/xmlmodule.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlmodule.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlreader.h: libxml/xmlreader.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlreader.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlregexp.h: libxml/xmlregexp.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlregexp.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlsave.h: libxml/xmlsave.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlsave.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlschemas.h: libxml/xmlschemas.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlschemas.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlschemastypes.h: libxml/xmlschemastypes.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlschemastypes.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlstring.h: libxml/xmlstring.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlstring.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlunicode.h: libxml/xmlunicode.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlunicode.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlversion.h: libxml/xmlversion.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlversion.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlwriter.h: libxml/xmlwriter.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlwriter.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xpath.h: libxml/xpath.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xpath.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xpathInternals.h: libxml/xpathInternals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xpathInternals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xpointer.h: libxml/xpointer.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xpointer.in.h; \
 	} > $@-t
diff -r 42fc00f7cdcb gettext-tools/gnulib-lib/Makefile.in
--- a/gettext-tools/gnulib-lib/Makefile.in	Mon Aug 06 02:08:14 2012 -0700
+++ b/gettext-tools/gnulib-lib/Makefile.in	Mon Aug 06 02:09:11 2012 -0700
@@ -2598,371 +2598,371 @@
 	mv -f $@-t $@
 
 glib/ghash.h: glib/ghash.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/ghash.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/glist.h: glib/glist.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/glist.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gprimes.h: glib/gprimes.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gprimes.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gstrfuncs.h: glib/gstrfuncs.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gstrfuncs.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gstring.h: glib/gstring.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gstring.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gtypes.h: glib/gtypes.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gtypes.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/DOCBparser.h: libxml/DOCBparser.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/DOCBparser.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/HTMLparser.h: libxml/HTMLparser.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/HTMLparser.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/HTMLtree.h: libxml/HTMLtree.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/HTMLtree.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/SAX.h: libxml/SAX.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/SAX.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/SAX2.h: libxml/SAX2.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/SAX2.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/c14n.h: libxml/c14n.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/c14n.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/catalog.h: libxml/catalog.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/catalog.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/chvalid.h: libxml/chvalid.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/chvalid.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/debugXML.h: libxml/debugXML.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/debugXML.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/dict.h: libxml/dict.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/dict.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/encoding.h: libxml/encoding.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/encoding.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/entities.h: libxml/entities.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/entities.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/globals.h: libxml/globals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/globals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/hash.h: libxml/hash.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/hash.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/list.h: libxml/list.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/list.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/nanoftp.h: libxml/nanoftp.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/nanoftp.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/nanohttp.h: libxml/nanohttp.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/nanohttp.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/parser.h: libxml/parser.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/parser.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/parserInternals.h: libxml/parserInternals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/parserInternals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/pattern.h: libxml/pattern.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/pattern.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/relaxng.h: libxml/relaxng.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/relaxng.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/schemasInternals.h: libxml/schemasInternals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/schemasInternals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/schematron.h: libxml/schematron.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/schematron.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/threads.h: libxml/threads.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/threads.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/tree.h: libxml/tree.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/tree.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/uri.h: libxml/uri.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/uri.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/valid.h: libxml/valid.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/valid.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xinclude.h: libxml/xinclude.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xinclude.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xlink.h: libxml/xlink.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xlink.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlIO.h: libxml/xmlIO.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlIO.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlautomata.h: libxml/xmlautomata.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlautomata.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlerror.h: libxml/xmlerror.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlerror.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlexports.h: libxml/xmlexports.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlexports.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlmemory.h: libxml/xmlmemory.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlmemory.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlmodule.h: libxml/xmlmodule.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlmodule.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlreader.h: libxml/xmlreader.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlreader.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlregexp.h: libxml/xmlregexp.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlregexp.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlsave.h: libxml/xmlsave.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlsave.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlschemas.h: libxml/xmlschemas.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlschemas.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlschemastypes.h: libxml/xmlschemastypes.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlschemastypes.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlstring.h: libxml/xmlstring.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlstring.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlunicode.h: libxml/xmlunicode.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlunicode.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlversion.h: libxml/xmlversion.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlversion.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xmlwriter.h: libxml/xmlwriter.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xmlwriter.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xpath.h: libxml/xpath.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xpath.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xpathInternals.h: libxml/xpathInternals.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xpathInternals.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 libxml/xpointer.h: libxml/xpointer.in.h
-	test -d libxml || mkdir libxml
+	mkdir -p libxml
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/libxml/xpointer.in.h; \
 	} > $@-t
diff -r 42fc00f7cdcb gnulib-local/modules/libglib
--- a/gnulib-local/modules/libglib	Mon Aug 06 02:08:14 2012 -0700
+++ b/gnulib-local/modules/libglib	Mon Aug 06 02:09:11 2012 -0700
@@ -58,42 +58,42 @@
 	mv -f $@-t $@
 
 glib/ghash.h: glib/ghash.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/ghash.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/glist.h: glib/glist.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/glist.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gprimes.h: glib/gprimes.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gprimes.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gstrfuncs.h: glib/gstrfuncs.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gstrfuncs.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gstring.h: glib/gstring.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gstring.in.h; \
 	} > $@-t
 	mv -f $@-t $@
 
 glib/gtypes.h: glib/gtypes.in.h
-	test -d glib || mkdir glib
+	mkdir -p glib
 	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
 	  cat $(srcdir)/glib/gtypes.in.h; \
 	} > $@-t
