diff -r c5bef1f432f6 Makefile.in
--- a/Makefile.in	Sat Jan 08 17:21:36 2011 -0800
+++ b/Makefile.in	Sat Jan 08 17:24:06 2011 -0800
@@ -419,7 +419,7 @@
 	gio-2.0-uninstalled.pc.in gio-unix-2.0-uninstalled.pc.in
 TEST_PROGS = 
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
-SUBDIRS = . m4macros glib gmodule gthread gobject gio tests po docs
+SUBDIRS = . glib gmodule gthread gobject gio po
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
diff -r c5bef1f432f6 gio/Makefile.in
--- a/gio/Makefile.in	Sat Jan 08 17:21:36 2011 -0800
+++ b/gio/Makefile.in	Sat Jan 08 17:24:06 2011 -0800
@@ -603,7 +603,7 @@
 TEST_PROGS = 
 NULL = 
 SUBDIRS = $(am__append_1) $(am__append_3) $(am__append_7) \
-	$(am__append_10) . $(am__append_13) tests
+	$(am__append_10) . $(am__append_13)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
diff -r c5bef1f432f6 glib/Makefile.in
--- a/glib/Makefile.in	Sat Jan 08 17:21:36 2011 -0800
+++ b/glib/Makefile.in	Sat Jan 08 17:24:06 2011 -0800
@@ -543,7 +543,7 @@
 @ENABLE_REGEX_TRUE@gregex_h = gregex.h
 @HAVE_GCC_BUILTINS_FOR_ATOMIC_OPERATIONS_FALSE@gatomic_c = gatomic.c
 @HAVE_GCC_BUILTINS_FOR_ATOMIC_OPERATIONS_TRUE@gatomic_c = gatomic-gcc.c
-SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre . tests
+SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre .
 DIST_SUBDIRS = libcharset gnulib pcre update-pcre tests
 AM_CPPFLAGS = \
 	$(glib_INCLUDES) 		\
diff -r c5bef1f432f6 gobject/Makefile.in
--- a/gobject/Makefile.in	Sat Jan 08 17:21:36 2011 -0800
+++ b/gobject/Makefile.in	Sat Jan 08 17:24:06 2011 -0800
@@ -437,7 +437,7 @@
 # GObject - GLib Type, Object, Parameter and Signal Library
 # Copyright (C) 1997,98,99,2000 Tim Janik and Red Hat, Inc.
 #
-SUBDIRS = . tests
+SUBDIRS = .
 CLEANFILES = $(am__append_2) $(gen_sources)
 @HAVE_THREADS_TRUE@THREAD_FLAGS = -DG_THREADS_MANDATORY
 AM_CPPFLAGS = \
diff -r c5bef1f432f6 gthread/Makefile.in
--- a/gthread/Makefile.in	Sat Jan 08 17:21:36 2011 -0800
+++ b/gthread/Makefile.in	Sat Jan 08 17:24:06 2011 -0800
@@ -394,7 +394,7 @@
 EXTRA_DIST = makefile.msc.in gthread-posix.c gthread-win32.c \
 	gthread-none.c gthread.def gthread.rc.in
 TEST_PROGS = 
-SUBDIRS = . tests
+SUBDIRS = .
 DIST_SUBDIRS = tests
 AM_CPPFLAGS = \
 	$(glib_INCLUDES) 		\
