diff -r fa44dc5b2fb4 Makefile.in
--- a/Makefile.in	Wed Mar 31 22:37:04 2010 -0700
+++ b/Makefile.in	Wed Mar 31 22:39:59 2010 -0700
@@ -403,7 +403,7 @@
 	gio-2.0-uninstalled.pc.in gio-unix-2.0-uninstalled.pc.in
 TEST_PROGS = 
 AUTOMAKE_OPTIONS = 1.7
-SUBDIRS = . m4macros glib gmodule gthread gobject gio tests po docs
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
diff -r fa44dc5b2fb4 gio/Makefile.in
--- a/gio/Makefile.in	Wed Mar 31 22:37:04 2010 -0700
+++ b/gio/Makefile.in	Wed Mar 31 22:39:59 2010 -0700
@@ -511,7 +511,7 @@
 TEST_PROGS = 
 NULL = 
 SUBDIRS = $(am__append_1) $(am__append_2) $(am__append_6) \
-	$(am__append_9) . $(am__append_12) tests
+	$(am__append_9) . $(am__append_12)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
diff -r fa44dc5b2fb4 glib/Makefile.in
--- a/glib/Makefile.in	Wed Mar 31 22:37:04 2010 -0700
+++ b/glib/Makefile.in	Wed Mar 31 22:39:59 2010 -0700
@@ -501,7 +501,7 @@
 @ENABLE_REGEX_TRUE@gregex_h = gregex.h
 @HAVE_GCC_BUILTINS_FOR_ATOMIC_OPERATIONS_FALSE@gatomic_c = gatomic.c
 @HAVE_GCC_BUILTINS_FOR_ATOMIC_OPERATIONS_TRUE@gatomic_c = gatomic-gcc.c
-SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre . tests
+SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre .
 DIST_SUBDIRS = libcharset gnulib pcre update-pcre tests
 AM_CPPFLAGS = \
 	-I$(top_srcdir) 		\
diff -r fa44dc5b2fb4 gobject/Makefile.in
--- a/gobject/Makefile.in	Wed Mar 31 22:37:04 2010 -0700
+++ b/gobject/Makefile.in	Wed Mar 31 22:39:59 2010 -0700
@@ -417,7 +417,7 @@
 # GObject - GLib Type, Object, Parameter and Signal Library
 # Copyright (C) 1997,98,99,2000 Tim Janik and Red Hat, Inc.
 #
-SUBDIRS = . tests
+SUBDIRS = .
 @HAVE_THREADS_TRUE@THREAD_FLAGS = -DG_THREADS_MANDATORY
 AM_CPPFLAGS = \
 	-DG_LOG_DOMAIN=\"GLib-GObject\"		\
