diff -r e952e5950603 src/Makefile.in
--- a/src/Makefile.in	Sat Dec 04 16:06:10 2010 -0800
+++ b/src/Makefile.in	Sat Dec 04 16:07:34 2010 -0800
@@ -867,7 +867,7 @@
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
 
 # Tell the linker to omit references to unused shared libraries.
-AM_LDFLAGS = $(IGNORE_UNUSED_LIBRARIES_CFLAGS)
+AM_LDFLAGS = $(IGNORE_UNUSED_LIBRARIES_CFLAGS) -mwindows
 grep_SOURCES = grep.c
 egrep_SOURCES = egrep.c
 fgrep_SOURCES = fgrep.c
diff -r e952e5950603 src/kwset.c
--- a/src/kwset.c	Sat Dec 04 16:06:10 2010 -0800
+++ b/src/kwset.c	Sat Dec 04 16:07:34 2010 -0800
@@ -31,11 +31,6 @@
 
 #include <config.h>
 #include <sys/types.h>
-#include "system.h"
-#include "kwset.h"
-#include "obstack.h"
-
-#define link kwset_link
 
 #ifdef GREP
 # include "xalloc.h"
@@ -43,6 +38,12 @@
 # define malloc xmalloc
 #endif
 
+#include "system.h"
+#include "kwset.h"
+#include "obstack.h"
+
+#define link kwset_link
+
 #define NCHAR (UCHAR_MAX + 1)
 #define obstack_chunk_alloc malloc
 #define obstack_chunk_free free
