diff -r 742e535cb0b5 Makefile.in
--- a/Makefile.in	Mon Mar 29 02:35:33 2010 -0700
+++ b/Makefile.in	Mon Mar 29 03:21:20 2010 -0700
@@ -469,8 +469,8 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
-SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos
+SUBDIRS = po po-properties $(SRC_SUBDIRS) m4macros
 
 # require automake 1.4
 AUTOMAKE_OPTIONS = 1.7
diff -r 742e535cb0b5 gdk/Makefile.in
--- a/gdk/Makefile.in	Mon Mar 29 02:35:33 2010 -0700
+++ b/gdk/Makefile.in	Mon Mar 29 03:21:20 2010 -0700
@@ -515,7 +515,7 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SUBDIRS = $(gdktarget) . tests
+SUBDIRS = $(gdktarget) .
 DIST_SUBDIRS = win32 x11 quartz directfb tests
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"Gdk\"		\
diff -r 742e535cb0b5 gtk/Makefile.in
--- a/gtk/Makefile.in	Mon Mar 29 02:35:33 2010 -0700
+++ b/gtk/Makefile.in	Mon Mar 29 03:21:20 2010 -0700
@@ -937,7 +937,7 @@
 
 # Makefile.am for gtk+/gtk
 @USE_QUARTZ_TRUE@GTK_PRINT_PREVIEW_COMMAND = "open -a /Applications/Preview.app %f"
-SUBDIRS = theme-bits . tests
+SUBDIRS = theme-bits .
 DIST_SUBDIRS = theme-bits tests
 @HAVE_CUPS_FALSE@@HAVE_PAPI_CUPS_FALSE@@HAVE_PAPI_FALSE@GTK_PRINT_BACKENDS = file,lpr
 @HAVE_CUPS_FALSE@@HAVE_PAPI_CUPS_FALSE@@HAVE_PAPI_TRUE@GTK_PRINT_BACKENDS = file,papi
diff -r 742e535cb0b5 modules/other/gail/Makefile.in
--- a/modules/other/gail/Makefile.in	Mon Mar 29 02:35:33 2010 -0700
+++ b/modules/other/gail/Makefile.in	Mon Mar 29 03:21:20 2010 -0700
@@ -456,7 +456,7 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SUBDIRS = libgail-util tests
+SUBDIRS = libgail-util
 @PLATFORM_WIN32_TRUE@no_undefined = -no-undefined
 moduledir = $(libdir)/gtk-2.0/modules
 module_LTLIBRARIES = libgail.la
diff -r 3f89048b07ef gdk/win32/gdkwindow-win32.c
--- a/gdk/win32/gdkwindow-win32.c	Sun Apr 18 11:50:14 2010 -0700
+++ b/gdk/win32/gdkwindow-win32.c	Sun Apr 18 12:13:54 2010 -0700
@@ -1128,7 +1128,7 @@
     {
       ShowWindow (GDK_WINDOW_HWND (window), SW_SHOWNOACTIVATE);
     }
-  else
+  else if (!IsWindowVisible (GDK_WINDOW_HWND (window)))
     {
       ShowWindow (GDK_WINDOW_HWND (window), SW_SHOWNORMAL);
     }
@@ -3585,6 +3585,8 @@
 gdk_window_focus (GdkWindow *window,
                   guint32    timestamp)
 {
+  GdkWindowObject *private = (GdkWindowObject *) window;
+
   g_return_if_fail (GDK_IS_WINDOW (window));
 
   if (GDK_WINDOW_DESTROYED (window))
@@ -3594,10 +3596,13 @@
 			   GDK_WINDOW_HWND (window),
 			   _gdk_win32_window_state_to_string (((GdkWindowObject *) window)->state)));
 
-  if (((GdkWindowObject *) window)->state & GDK_WINDOW_STATE_MAXIMIZED)
+  if (private->state & GDK_WINDOW_STATE_MAXIMIZED)
     ShowWindow (GDK_WINDOW_HWND (window), SW_SHOWMAXIMIZED);
-  else
+  else if (private->state & GDK_WINDOW_STATE_ICONIFIED)
+    ShowWindow (GDK_WINDOW_HWND (window), SW_RESTORE);
+  else if (!IsWindowVisible (GDK_WINDOW_HWND (window)))
     ShowWindow (GDK_WINDOW_HWND (window), SW_SHOWNORMAL);
+
   SetFocus (GDK_WINDOW_HWND (window));
 }
 
