diff -r e189f017f0e8 Makefile.in
--- a/Makefile.in	Mon Mar 29 01:45:29 2010 -0700
+++ b/Makefile.in	Mon Mar 29 01:49:15 2010 -0700
@@ -1236,8 +1236,9 @@
 	cd $(DESTDIR)$(includedir); $(LN_S) $(PNGLIB_BASENAME)/png.h png.h
 	cd $(DESTDIR)$(includedir); $(LN_S) $(PNGLIB_BASENAME)/pngconf.h \
 		pngconf.h
-	cd $(DESTDIR)$(pkgconfigdir); rm -f libpng.pc
+	cd $(DESTDIR)$(pkgconfigdir); rm -f libpng.pc libpng12.pc
 	cd $(DESTDIR)$(pkgconfigdir); $(LN_S) $(PNGLIB_BASENAME).pc libpng.pc
+	cd $(DESTDIR)$(pkgconfigdir); $(LN_S) $(PNGLIB_BASENAME).pc libpng12.pc
 
 # do evil things to libpng to cause libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@ to be used
 install-exec-hook:
@@ -1255,6 +1256,7 @@
 uninstall-hook:
 	cd $(DESTDIR)$(includedir); rm -f png.h pngconf.h
 	rm -f $(DESTDIR)$(pkgconfigdir)/libpng.pc
+	rm -f $(DESTDIR)$(pkgconfigdir)/libpng12.pc
 	rm -f $(DESTDIR)$(bindir)/libpng-config
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -r e189f017f0e8 configure
--- a/configure	Mon Mar 29 01:45:29 2010 -0700
+++ b/configure	Mon Mar 29 01:49:15 2010 -0700
@@ -11438,6 +11438,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: *** You have not enabled versioned symbols." >&5
 $as_echo "$as_me: WARNING: *** You have not enabled versioned symbols." >&2;}
 fi
+have_ld_version_script=no
  if test "$have_ld_version_script" = "yes"; then
   HAVE_LD_VERSION_SCRIPT_TRUE=
   HAVE_LD_VERSION_SCRIPT_FALSE='#'
