diff -r e32d18cbb6e3 Makefile
--- a/Makefile	Wed Sep 08 23:31:13 2010 -0700
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,5 +0,0 @@
-all:
-	-@echo "Please use ./configure first.  Thank you."
-
-distclean:
-	make -f Makefile.in distclean
diff -r e32d18cbb6e3 Makefile.in
--- a/Makefile.in	Wed Sep 08 23:31:13 2010 -0700
+++ b/Makefile.in	Wed Sep 08 23:33:15 2010 -0700
@@ -39,7 +39,7 @@
 AR=ar rc
 RANLIB=ranlib
 LDCONFIG=ldconfig
-LDSHAREDLIBC=-lc
+LDSHAREDLIBC=
 TAR=tar
 SHELL=/bin/sh
 EXE=
@@ -67,7 +67,7 @@

 PIC_OBJS = $(PIC_OBJC) $(PIC_OBJA)

-all: static shared
+all: dll

 static: example$(EXE) minigzip$(EXE)

@@ -119,7 +119,7 @@

 match.lo: match.S
 	$(CPP) match.S > _match.s
-	$(CC) -c -fPIC _match.s
+	$(CC) -c _match.s
 	mv _match.o match.lo
 	rm -f _match.s

@@ -185,12 +185,12 @@
 # The ranlib in install is needed on NeXTSTEP which checks file times
 # ldconfig is for Linux

-install: install-libs
+install: install-dll
 	-@if [ ! -d $(DESTDIR)$(includedir)   ]; then mkdir -p $(DESTDIR)$(includedir); fi
 	cp zlib.h zconf.h $(DESTDIR)$(includedir)
 	chmod 644 $(DESTDIR)$(includedir)/zlib.h $(DESTDIR)$(includedir)/zconf.h

-uninstall:
+uninstall: uninstall-dll
 	cd $(DESTDIR)$(includedir); rm -f zlib.h zconf.h
 	cd $(DESTDIR)$(libdir); rm -f libz.a; \
 	if test "$(SHAREDLIBV)" -a -f $(SHAREDLIBV); then \
@@ -234,6 +234,23 @@
 depend:
 	makedepend -- $(CFLAGS) -- *.[ch]

+# begin muntyan additions
+dlldir = $(exec_prefix)/bin
+DLL = zlib1-m314.dll
+DLL_A = libz.dll.a
+$(DLL): $(OBJS)
+	$(CC) -shared -o $(DLL) -Wl,--out-implib=$(DLL_A) $(OBJS)
+dll: $(DLL)
+install-dll: dll
+	-@if [ ! -d $(DESTDIR)$(dlldir) ]; then mkdir -p $(DESTDIR)$(dlldir); fi
+	-@if [ ! -d $(DESTDIR)$(libdir) ]; then mkdir -p $(DESTDIR)$(libdir); fi
+	cp $(DLL) $(DESTDIR)$(dlldir)
+	cp $(DLL_A) $(DESTDIR)$(libdir)
+	rm -f $(DESTDIR)$(libdir)/$(STATICLIB)
+uninstall-dll:
+	rm -f $(DESTDIR)$(dlldir)/$(DLL) $(DESTDIR)$(libdir)/$(DLL_A)
+# end muntyan additions
+
 # DO NOT DELETE THIS LINE -- make depend depends on it.

 adler32.o zutil.o: zutil.h zlib.h zconf.h
diff -r e32d18cbb6e3 configure
--- a/configure	Wed Sep 08 23:31:13 2010 -0700
+++ b/configure	Wed Sep 08 23:33:15 2010 -0700
@@ -105,7 +105,7 @@

 if test "$gcc" -eq 1 && ($cc -c $cflags $test.c) 2>/dev/null; then
   CC="$cc"
-  SFLAGS="${CFLAGS--O3} -fPIC"
+  SFLAGS="${CFLAGS--O3}"
   CFLAGS="${CFLAGS--O3}"
   if test $build64 -eq 1; then
     CFLAGS="${CFLAGS} -m64"
@@ -568,7 +568,6 @@
 /^sharedlibdir *=/s#=.*#=$sharedlibdir#
 /^includedir *=/s#=.*#=$includedir#
 /^mandir *=/s#=.*#=$mandir#
-/^all: */s#:.*#: $ALL#
 /^test: */s#:.*#: $TEST#
 " > Makefile

