diff -r d544ca93ff86 Makefile
--- a/Makefile	Sun Mar 28 23:33:00 2010 -0700
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,5 +0,0 @@
-all:
-	-@echo "Please use ./configure first.  Thank you."
-
-distclean:
-	make -f Makefile.in distclean
diff -r d544ca93ff86 Makefile.in
--- a/Makefile.in	Sun Mar 28 23:33:00 2010 -0700
+++ b/Makefile.in	Mon Mar 29 00:04:32 2010 -0700
@@ -64,7 +64,7 @@
 
 PIC_OBJS = $(PIC_OBJC) $(PIC_OBJA)
 
-all: static shared
+all: static shared dll
 
 static: example$(EXE) minigzip$(EXE)
 
@@ -116,7 +116,7 @@
 
 match.lo: match.S
 	$(CPP) match.S > _match.s
-	$(CC) -c -fPIC _match.s
+	$(CC) -c _match.s
 	mv _match.o match.lo
 	rm -f _match.s
 
@@ -134,7 +134,7 @@
 	-@mv objs/$*.o $@
 
 $(SHAREDLIBV): $(PIC_OBJS)
-	$(LDSHARED) $(SFLAGS) -o $@ $(PIC_OBJS) -lc
+	$(LDSHARED) $(SFLAGS) -o $@ $(PIC_OBJS)
 	rm -f $(SHAREDLIB) $(SHAREDLIBM)
 	ln -s $@ $(SHAREDLIB)
 	ln -s $@ $(SHAREDLIBM)
@@ -180,12 +180,12 @@
 # The ranlib in install is needed on NeXTSTEP which checks file times
 # ldconfig is for Linux
 
-install: install-libs
+install: install-libs install-dll
 	-@if [ ! -d $(DESTDIR)$(includedir)   ]; then mkdir -p $(DESTDIR)$(includedir); fi
 	cp zlib.h zconf.h $(DESTDIR)$(includedir)
 	chmod 644 $(DESTDIR)$(includedir)/zlib.h $(DESTDIR)$(includedir)/zconf.h
 
-uninstall:
+uninstall: uninstall-dll
 	cd $(DESTDIR)$(includedir); rm -f zlib.h zconf.h
 	cd $(DESTDIR)$(libdir); rm -f libz.a; \
 	if test "$(SHAREDLIBV)" -a -f $(SHAREDLIBV); then \
@@ -229,6 +229,23 @@
 depend:
 	makedepend -- $(CFLAGS) -- *.[ch]
 
+# begin muntyan additions
+dlldir = $(exec_prefix)/bin
+DLL = zlib1-314.dll
+DLL_A = libz.dll.a
+$(DLL): $(OBJS)
+	$(CC) -shared -o $(DLL) -Wl,--out-implib=$(DLL_A) $(OBJS)
+dll: $(DLL)
+install-dll: dll
+	-@if [ ! -d $(DESTDIR)$(dlldir) ]; then mkdir -p $(DESTDIR)$(dlldir); fi
+	-@if [ ! -d $(DESTDIR)$(libdir) ]; then mkdir -p $(DESTDIR)$(libdir); fi
+	cp $(DLL) $(DESTDIR)$(dlldir)
+	cp $(DLL_A) $(DESTDIR)$(libdir)
+	rm -f $(DESTDIR)$(libdir)/$(STATICLIB)
+uninstall-dll:
+	rm -f $(DESTDIR)$(dlldir)/$(DLL) $(DESTDIR)$(libdir)/$(DLL_A)
+# end muntyan additions
+
 # DO NOT DELETE THIS LINE -- make depend depends on it.
 
 adler32.o zutil.o: zutil.h zlib.h zconf.h
diff -r d544ca93ff86 configure
--- a/configure	Sun Mar 28 23:33:00 2010 -0700
+++ b/configure	Mon Mar 29 00:04:32 2010 -0700
@@ -78,7 +78,7 @@
 
 if test "$gcc" -eq 1 && ($cc -c $cflags $test.c) 2>/dev/null; then
   CC="$cc"
-  SFLAGS="${CFLAGS-"-O3"} -fPIC"
+  SFLAGS="${CFLAGS-"-O3"}"
   CFLAGS="${CFLAGS-"-O3"}"
   if test $build64 -eq 1; then
     CFLAGS="${CFLAGS} -m64"
